apiserver/facades/agent/unitassigner/unitassigner.go

```diff
@@ -21,17 +21,23 @@ import (
 type assignerState interface {
 	WatchForUnitAssignment() state.StringsWatcher
 	AssignStagedUnits(ids []string) ([]state.UnitAssignmentResult, error)
+	AssignedMachineId(unit string) (string, error)
 }
 
 type statusSetter interface {
 	SetStatus(context.Context, params.SetStatus) (params.ErrorResults, error)
 }
 
+type machineSaver interface {
+	Save(context.Context, string) error
```

Comment:
I'd prefer `AddMachine`, `UpdateMachine` and `DeleteMachine`. They will have different validation rules.